src/main/java/io/github/syst3ms/skriptparser/expressions/ExprStringCharAt.java

```diff
@@ -0,0 +1,52 @@
+package io.github.syst3ms.skriptparser.expressions;
+
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import io.github.syst3ms.skriptparser.Parser;
+import io.github.syst3ms.skriptparser.lang.Expression;
+import io.github.syst3ms.skriptparser.lang.TriggerContext;
+import io.github.syst3ms.skriptparser.parsing.ParseContext;
+
+/**
+ * The character at a position of a string. The first character is at the position 0.
+ *
+ * @name Character at
+ * @pattern char[acter] at [(index|pos[ition])] %number% (of|in) %string%
+ * @since ALPHA
+ * @author Olyno
+ */
+public class ExprStringCharAt implements Expression<String> {
+
+	static {
+		Parser.getMainRegistration().addExpression(
+			ExprStringCharAt.class,
+			String.class,
+			true,
+			"char[acter] at [(index|pos[ition])] %number% (of|in) %string%"
```

Comment:
Use `%integer%`. Yes, the type behind it is Long, and you have to convert it using `intValue()` anyway, but it will at least filter out users doing dumb stuff like inputting a decimal number (in fact, since they can, you have to expect that they will).